cmd/controller/main.go

```diff
@@ -41,7 +44,17 @@ import (
 
 func main() {
 	flag.IntVar(&controller.DefaultThreadsPerController, "threads-per-controller", controller.DefaultThreadsPerController, "Threads (goroutines) to create per controller")
-	namespace := flag.String("namespace", "", "Namespace to restrict informer to. Optional, defaults to all namespaces.")
+	namespaceList := flag.String("namespaces", "", "Comma-separated list of namespaces to restrict informer to. Optional, if empty defaults to all namespaces.")
+	flag.Parse()
+
+	var namespaces []string
+	if namespaceList != nil && *namespaceList != "" {
+		space := regexp.MustCompile(`\s+`)
+		namespaces = strings.Split(space.ReplaceAllString(*namespaceList, ""), ",")
+		fmt.Printf("controller is scopped to the following namespaces: %s\n", namespaces)
+	} else {
+		namespaces = nil // Default to all namespaces if the list is empty
```

Comment:
I think this will fail and won't set to all namespaces as `""` won't be set i.e. it will not default to all namespaces and it will fail here https://github.com/tektoncd/chains/pull/1148/files#diff-4ce7e773e47a412c0fcdc0db37ed953113e7744b300b2db1f485a20ca6b7ed0bR32 as it will have an empty array

Reply:
@PuneetPunamiya Why would it fail ? len(nil slice) = 0, and namespaces will not be an empty error because we force to nil fi the namespaces flag is not set or empty

Reply:
This seems unnecessary though. If it got here, it's already `nil`.